react/components/Newsletter/index.js

```diff
@@ -0,0 +1,173 @@
+import React, { Component, Fragment } from 'react'
+import { compose, graphql } from 'react-apollo'
+import PropTypes from 'prop-types'
+import { injectIntl, intlShape } from 'react-intl'
+import { Input, Button } from 'vtex.styleguide'
+import SUBSCRIBE_NEWSLETTER from './mutations/subscribeNewsletter.graphql'
+import style from './style.css'
+
+const EMAIL_REGEX = /^[A-z0-9+_-]+(?:\.[A-z0-9+_-]+)*@(?:[A-z0-9](?:[A-z0-9-]*[A-z0-9])?\.)+[A-z0-9](?:[A-z0-9-]*[A-z0-9])?$/
+
+class Newsletter extends Component {
+  state = {
+    email: '',
+    loading: false,
+    error: null,
+    success: null,
+    invalidEmail: false,
+  }
+
+  inputRef = React.createRef()
+
+  componentDidMount() {
+    this.mounted = true
+  }
+
+  componentWillUnmount() {
+    this.mounted = false
+  }
+
+  safeSetState(...params) {
+    if (this.mounted) {
+      this.setState(...params)
+    }
+  }
+
+  handleChangeEmail = (e) => {
+    this.setState({ email: e.target.value })
+  }
+
+  validateEmail = () => {
+    return EMAIL_REGEX.test(this.state.email)
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault()
+    if (!this.validateEmail()) {
+      this.setState({ invalidEmail: true })
+      if (this.inputRef && this.inputRef.current) {
+        this.inputRef.current.focus()
+      }
+      return
+    }
+
+    this.setState({ invalidEmail: false, loading: true, error: null, success: null })
+
+    this.props.subscribeNewsletter({ variables: { email: this.state.email } })
+      .then(() => {
+        this.safeSetState({ success: true, loading: false })
+      })
+      .catch((e) => {
+        this.safeSetState({ error: true, loading: false })
+      })
+  }
+
+  render() {
+    const {
+      placeholder = this.props.intl.formatMessage({ id: 'newsletter.placeholder' }),
+      submit = this.props.intl.formatMessage({ id: 'newsletter.submit' }),
+      label = this.props.intl.formatMessage({ id: 'newsletter.label' }),
+      hideLabel,
+    } = this.props
+
+    return (
+      <div className={`${style.newsletter} ${this.state.success ? style.confirmation : ''} w-100`}>
+        <div className={`${style.container} mw9 mr-auto ml-auto pv9`}>
+          {this.state.success
+            ? (
+              <Fragment>
+                <div className={`${style.confirmationTitle} t-heading-3 pb4 tc`}>
+                  {this.props.intl.formatMessage({ id: 'newsletter.confirmationTitle' })}
```

Comment:
You could use `<FormattedMessage>` component

Reply:
True